app/migrations.go

```diff
@@ -30,27 +30,31 @@ func (a *App) DoAdvancedPermissionsMigration() {
 	allSucceeded := true
 
 	for _, role := range roles {
-		if result := <-a.Srv.Store.Role().Save(role); result.Err != nil {
-			// If this failed for reasons other than the role already existing, don't mark the migration as done.
-			if result2 := <-a.Srv.Store.Role().GetByName(role.Name); result2.Err != nil {
+		_, err := a.Srv.Store.Role().Save(role)
+		if err == nil {
+			continue
+		}
+
+		// If this failed for reasons other than the role already existing, don't mark the migration as done.
+		fetchedRole, err := a.Srv.Store.Role().GetByName(role.Name)
+		if err != nil {
+			mlog.Critical("Failed to migrate role to database.")
+			mlog.Critical(fmt.Sprint(err))
```

Comment:
Since we're changing this line, can we just fold it into the one above with as follows:
```go
			mlog.Critical("Failed to migrate role to database.", mlog.Err(err))
```

There are similar logs below where we're modifying the line as well.

Reply:
That makes a lot of sense, I going to do it.

Reply:
Done! PTAL to the file changes.